src/PriceOracle.ts

```diff
@@ -84,31 +85,30 @@ export async function set_whitelisted_prices(chainId: number, blockNumber: numbe
         .map(token => token.address);
 
     if (addresses.length === 0) return;
-    const units = tokenData.map(token => token.unit as validUnit);
     const prices = await read_prices(addresses, chainId, blockNumber);
 
     // Map prices to token addresses
-    const pricesByAddress = new Map<string, number>();
-    pricesByAddress.set(CHAIN_CONSTANTS[chainId].usdc.address, 1);
+    const pricesByAddress = new Map<string, string>();
+    pricesByAddress.set(CHAIN_CONSTANTS[chainId].usdc.address, "1");
```

Comment:
This is the file we going to have merge conflicts with the cache system I included in #137 - we should think about how we merge